File: wup1/Java/NotString.java
/* 
   notString

   Given a string, return a new string where "not " has been added to the front. 
   However, if the string already begins with "not", return the string unchanged. 
   Note: use .equals() to compare 2 strings.

*/

public class NotString {
    
   public static void main(String[] args) {
      NotString ns = new NotString();
      ns.test();
	}

   public String notString(String str) {
      if (str.length() >= 3)
         if (str.substring(0, 3).equals("not"))
            return str;
            
      return "not " + str;
   }

   public void test() {
      assert notString("candy").equals("not candy");
      assert notString("x").equals("not x");
      assert notString("not bad").equals("not bad");
      assert notString("bad").equals("not bad");
      assert notString("not").equals("not");
      assert notString("is not").equals("not is not");
      assert notString("no").equals("not no");
      System.out.println("All tests passed");
   }
}

File: wup1/JS/notString.js
/*
    notString

    Given a string, return a new string where "not " has been added to the front. 
    However, if the string already begins with "not", return the string unchanged.
*/

function notString(str){
    if (str.length >= 3 && str.substring(0,3) == "not") {
        return str;
    }
    return 'not ' + str;
}

const assert = require('assert');
assert.equal(notString('candy') , 'not candy');
assert.equal(notString('x'), 'not x');
assert.equal(notString('not bad'), 'not bad');
assert.equal(notString('bad'), 'not bad');	
assert.equal(notString('not'), 'not');	
assert.equal(notString('is not'), 'not is not');	
assert.equal(notString('no'), 'not no');

console.log("All tests passed");